api/controller/assignments.js

```diff
@@ -0,0 +1,904 @@
+                                },
+                                {
+                                    $project: {
+                                        'content.objectId': 1,
+                                        'content.metadata': 1,
+                                        'content.text': 1,
+                                    }
+                                }
+                            ).toArray(function (err, journals) {
+                                if (err) {
+                                    return res.status(500).send({
+                                        'error': "An error has occurred",
+                                        'code': 10
+                                    });
+                                }
+                                journals.find(function (journal) {
+                                    journal.content.filter(function (entry) {
+                                        if (entry.objectId === assignment.assignedWork) {
+                                            assignment.assignedWork = entry;
+                                        }
+                                    });
+                                });
+                                res.status(200).send(assignment);
+                            });
+                        });
+                    }
+                });
+            });
+        });
+    });
+};
+
+//launch Assignment
+exports.launchAssignment = function (req, res) {
+    //validate
+    if (!mongo.ObjectID.isValid(req.params.assignmentId)) {
+        return res.status(401).send({
+            'error': "Invalid assignment id",
+            'code': 35
+        });
+    }
+    //find assignment by id
+    db.collection(assignmentCollection, function (err, collection) {
+        collection.findOne({ _id: new mongo.ObjectID(req.params.assignmentId) }, function (err, assignment) {
+            if (err) {
+                return res.status(400).send({
+                    'error': 'Inexisting assignment id',
+                    'code': 23
+                });
+            }
+            if (!assignment) {
+                return res.status(401).send({});
+            }
+            else {
+                //find all students from classrooms
+                var classrooms = assignment.classrooms;
+                var privateJournalIds = []; //to store private journal ids of every student
+                //get students private_journal
+                common.fetchAllStudents(classrooms).then(function (stds) {
+                    var uniqueJournalIds = [];
+                    //make set for getting unique values of private_journal
+                    var uniqueJournalIdsSet = new Set();
+                    for (var i = 0; i < stds.length; i++) {
+                        if (!uniqueJournalIdsSet.has(stds[i].private_journal.toString())) {
+                            uniqueJournalIdsSet.add(stds[i].private_journal.toString());
+                        }
```

Comment:
Set can contain an element only once. So check for "has" is not needed. Maybe you can confirm this and update the code if needed.

Reply:
I checked this when I was doing this. It works fine in this way